cmd/argocd/commands/login.go

```diff
@@ -48,7 +48,9 @@ func NewLoginCommand(globalClientOpts *argocdclient.ClientOptions) *cobra.Comman
 				}
 			}
 
-			username, password = cli.PromptCredentials()
+			if username == "" || password == "" {
+				username, password = cli.PromptCredentials()
+			}
```

Comment:
Oops I broke this, yesterday. I think even this fix is not totally correct, because one of username or one of password could be supplied, and it should not prompt when it is.